bundles/org.eclipse.jface/src/org/eclipse/jface/resource/FileImageDescriptor.java

```diff
@@ -304,12 +303,24 @@ public String getImagePath(int zoom) {
 		return null;
 	}
 
+	private URL toURL() {
+		if (location == null) {
+			return null;
+		}
+
+		if (name == null) {
+			return null;
+		}
+
+		return location.getResource(name);
+	}
+
 	@Override
 	public <T> T getAdapter(Class<T> adapter) {
-		if (adapter == ImageFileNameProvider.class) {
-			// Support testing ImageFileNameProvider characteristics, see #396
-			return adapter.cast(this);
+		if (adapter == URL.class) {
+			return adapter.cast(toURL());
```

Comment:
If to URL is only used here, it might be better to inline the code than having a new method....

Reply:
I have no problem with that suggestion.

Reply:
Done.